pkg/agent/agent.go

```diff
@@ -1132,26 +1135,17 @@ func getRoundInfo(bridgeClient ovsconfig.OVSBridgeClient) types.RoundInfo {
 	return roundInfo
 }
 
-func (i *Initializer) getNodeMTU(transportInterface *net.Interface) (int, error) {
+func (i *Initializer) getInterfaceMTU(transportInterface *net.Interface) (int, error) {
 	if i.mtu != 0 {
 		return i.mtu, nil
 	}
-	mtu := transportInterface.MTU
+	mtu := transportInterface.MTU - i.networkConfig.MTUDeduction
 	// Make sure mtu is set on the interface.
 	if mtu <= 0 {
 		return 0, fmt.Errorf("Failed to fetch Node MTU : %v", mtu)
 	}
-	if i.networkConfig.TrafficEncapMode.SupportsEncap() {
-		if i.networkConfig.TunnelType == ovsconfig.VXLANTunnel {
-			mtu -= config.VXLANOverhead
-		} else if i.networkConfig.TunnelType == ovsconfig.GeneveTunnel {
-			mtu -= config.GeneveOverhead
-		} else if i.networkConfig.TunnelType == ovsconfig.GRETunnel {
-			mtu -= config.GREOverhead
-		}
-		if i.nodeConfig.NodeIPv6Addr != nil {
-			mtu -= config.IPv6ExtraOverhead
-		}
+	if i.nodeConfig.NodeIPv6Addr != nil {
```

Comment:
Please add the `i.networkConfig.TrafficEncapMode.SupportsEncap()` check back.

Reply:
Done